ui/account/src/main/java/app/tivi/account/AccountPresenter.kt

```diff
@@ -21,28 +21,47 @@ import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.collectAsState
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.rememberCoroutineScope
-import androidx.lifecycle.ViewModel
 import app.tivi.data.traktauth.LoginToTraktInteractor
 import app.tivi.data.traktauth.TraktAuthRepository
 import app.tivi.data.traktauth.TraktAuthState
 import app.tivi.domain.interactors.ClearUserDetails
 import app.tivi.domain.observers.ObserveTraktAuthState
 import app.tivi.domain.observers.ObserveUserDetails
+import app.tivi.screens.AccountScreen
+import app.tivi.screens.SettingsScreen
+import com.slack.circuit.runtime.CircuitContext
+import com.slack.circuit.runtime.Navigator
+import com.slack.circuit.runtime.Screen
+import com.slack.circuit.runtime.presenter.Presenter
 import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
+import me.tatarka.inject.annotations.Assisted
 import me.tatarka.inject.annotations.Inject
 
 @Inject
-class AccountUiViewModel(
+class AccountUiPresenterFactory(
+    private val presenterFactory: (Navigator) -> AccountPresenter,
+) : Presenter.Factory {
+    override fun create(screen: Screen, navigator: Navigator, context: CircuitContext): Presenter<*>? {
+        return when (screen) {
+            is AccountScreen -> presenterFactory(navigator)
+            else -> null
+        }
+    }
+}
```

Comment:
For cases not using a code gen tool, anything we could do to make these simpler? Maybe a `ScreenPresenterFactory(Screen,Function)` API that dries some of this up?

Reply:
I thought about writing exactly that, but the only thing you're really saving is the `when`.

You still need a class somewhere to inject the assisted factory. I guess you could move that to the component/module instead, and just call a `ScreenPresenterFactory()`.